packages/noir-ethereum-api/src/ethereum/createRecordingClient.ts

```diff
@@ -0,0 +1,51 @@
+import { PublicClient } from 'viem';
+
+export type CallResultEntry = {
+  method: string;
+  arguments: object[];
+  result: Promise<object>;
+};
+
+export type AwaitedCallResultEntry = {
+  method: string;
+  arguments: object[];
+  result: object;
+};
+
+export type GetCallResults = { getCallResults: () => Promise<AwaitedCallResultEntry[]> };
+
+export const createRecordingClient = (client: PublicClient): PublicClient & GetCallResults =>
+  createLoggingProxy(client);
```

Comment:
Why not inline - one line method?

Reply:
After extracting `propCondition` separation of creating proxy from creating recording client is much more clear.

Reply:
Now perfect